dowhy/datasets.py

```diff
@@ -109,6 +114,8 @@ def linear_dataset(beta, num_common_causes, num_samples, num_instruments=0,
     # Converting treatment to binary if required
     if treatment_is_binary:
         t = np.vectorize(stochastically_convert_to_binary)(t)
+    elif treatment_is_category:
+        t = np.vectorize(stochastically_convert_to_three_level_categorical)(t)
```

Comment:
I'm trying to understand if this line is enough to create a discrete treatment. With `num_treatments=1`,  this line should create a single vector with values in the set {0,1,2}. But for categorical treatment, we'd like to one-hot encode it, since it gets multiplied with beta in `_compute_y`.  Is the one-hot encoding already happening somehow?

Reply:
No, we do not do one-hot encoding within `datasets.py`. Instead, we user integer coding and we expect users to use integer coding for their datasets. Then `get_dummies` is called by us inside the regression estimation code

Reply:
See my other comment where we set the correct pandas type for this data

Reply:
Okay, I understand. One issue is that datasets.py simulates `y` based on the input "true" causal effect. Right now, when 'y` is computed, it is computed as a linear function of integer-encoded categorical `t`, which would be incorrect. So perhaps we need to do two things:
1) output the integer-encoded categorical `t` in the data frame, as you are currently doing
2) use one hot encoding on `t` inside `_compute_y` function before multiplying it with beta. 

This is important because downstream methods measure their accuracy wrt `beta` as the true causal effect. Actually, it makes me think: how should beta be interpreted for discrete treatments because it is no longer a singleton effect? One idea is to define `beta` as a list of effects of size equal to (num_categories-1). `beta[i]` is interpreted as the causal effect of treatment category `i` compared to control, essentially the linear regression coefficients for each one-hot encoded variable. Actually, since multiple treatments are allowed, `beta` will be a list of lists, e.g., if there are two treatments, 
one is discrete {0,1,2}, the other is continuous, then `beta=[[10,5], [10]]`. does this make sense?

Reply:
I think we can break this down into two different things: 1) is it ok to run `_compute_y` with `t` being integer coded, or is it necessary to one-hot encode it? 2) how do we interpret `beta` in the categorical regression model?

For 1), since `datasets.py` is creating an example dataset I think it is completely fine if the effect *just happens* to be linear in the integer coded treatments. This would not be true in general, but for our tests it seems reasonable and saves a refactor. On the other hand, implementing a nonlinear effect would be fine too!

For 2), I think we should focus on ATE for evaluation rather than `beta`. The ATE is well defined between any pair of treatments. The `beta` is sensitive to all kinds of things (variable scaling, whether we set the first row to be control or made a mistake, etc). For the new tests I added with this new dataset, you can see that we are able to recover the right ATE. If you set things correctly, so control is in the first row, then yes, your `beta` would be a list (or a list of lists) of the treatment effect sizes. But I really think ATE is better as it transfers to more complex models. E.g. what if you have two treatments, but now allow interactions, then the list of lists should include all pairs- it just becomes a bit confusing.

Reply:
okay, for 1), can you add a comment describing the point, so that future readers can refactor if needed. 
for 2): good point on focusing on ATE. I agree with its generality. So all good there.

Reply:
Thanks @amit-sharma ! I just pushed a new comment on the linearity assumption that is made for this test dataset